src/v/container/include/container/fragmented_vector.h

```diff
@@ -190,9 +208,52 @@ class fragmented_vector {
     size_t size() const noexcept { return _size; }
 
     void shrink_to_fit() {
-        if (!_frags.empty()) {
-            _frags.back().shrink_to_fit();
+        // Calling shrink to fix then modifying the container could result in
+        // allocations that overshoot the template parameter.
+        //
+        // if (!_frags.empty()) {
+        //    _frags.back().shrink_to_fit();
+        // }
+    }
+
+    void reserve(size_t v) {
+        // For fixed size fragments we noop, as we already reserve the full size
+        // of vector
+        if constexpr (is_chunked_vector) {
```

Comment:
Why don't we just make this method available only on the fixed-size vector if it does nothing in the dynamic variant?

Reply:
Do you mean to reverse that? How would propose to do this? Make a subclass for the dynamic variant? I'm honestly hoping that the non-dynamic version goes away eventually.

Reply:
Yes, I meant to reverse that. 

You could use SFINAE/enable_if but the easiest is probably just `static_assert`.

Reply:
(see suggestion above)

Reply:
> I'm honestly hoping that the non-dynamic version goes away eventually.

hard agree

Reply:
Added a static assert for now.